azure-pipelines.yml

```diff
@@ -1,170 +1,102 @@
-    steps:
-    - script: |
-        domain=`echo "$(MPURL)" | awk -F[/:] '{print $4}'`
-        if [[ "$(MPURL)" == *"-dev.buyingcatalogue.digital.nhs.uk"* ]]
-        then
-          ip="51.11.46.27"
-        elif [[ "$(MPURL)" == *"private-test.buyingcatalogue.digital.nhs.uk"* ]]
-        then
-          ip="51.132.54.44"
-        elif [[ "$(MPURL)" == *"public-test.buyingcatalogue.digital.nhs.uk"* ]]
-        then
-          ip="51.132.53.156"
-        fi
-        if [[ -n "$ip" ]]
-        then
-          echo "$ip        $domain" | sudo tee -a /etc/hosts
-          cat domain-template.yaml | sed "s/DOMAIN/$domain/g;s/IP/$ip/g" > domain.yml
-        else
-          touch domain.yml
-        fi
-        curl -k -i $(MPURL) --fail --connect-timeout 30
-        cat domain.yml
-        cat /etc/hosts
-      displayName: 'Resolve domain'
-    - task: UseDotNet@2
-      displayName: 'Use DotNet Core 3.1.x'
-      inputs:
-        packageType: 'sdk'
-        version: '3.1.101'
-        includePreviewVersions: false
-        installationPath: '$(Agent.ToolsDirectory)'
-
-    - task: AzureKeyVault@1
-      inputs:
-        azureSubscription: 'NHSAPP-BuyingCatalogue (Non-Prod)'
-        KeyVaultName: 'gpitfutures-dev-kv'
-        SecretsFilter: "*"
-
-    - task: A8515EC8-7254-4FFD-912C-86772E2B5962@3
-      displayName: 'Replace tokens'
-      inputs:
-        targetFiles: '**/*.json'
-        encoding: 'auto'
-        writeBOM: true
-        actionOnMissing: 'continue'
-        keepToken: true
-        tokenPrefix: '#{'
-        tokenSuffix: '}#'
-        emptyValue: 
-    
-    - task: Bash@3
-      displayName: Selenium Grid Setup
-      inputs:
-        targetType: 'inline'
-        script: |
-          docker-compose stop &&
-          docker-compose rm -f &&
-          docker-compose -f docker-compose.yml -f domain.yml up --scale chrome=4 --scale firefox=4  -d
-    - task: DotNetCoreCLI@2
-      displayName: "Run Authority Acceptance Tests"
-      inputs:
-        command: 'test'
-        projects: 'src/MarketingPageAcceptanceTests.AuthorityTests/*csproj'
-        arguments: '-v n'
\ No newline at end of file
+  - name: buildConfiguration
+    value: 'Release'
+  - name: dotnetVersion
+    value: '3.1.x'
+  - name: MSBUILDSINGLELOADCONTEXT
+    value: '1'
+  - name: BROWSER
+    value: chrome
+  - name: DB__SERVERURL
+    value: gpitfutures-dev-sql-pri.database.windows.net
+  - name: DB__DATABASENAME
+    value: bc-buyingcatalogue-development-helm-bapi
+  - name: AZUREBLOBSTORAGE__CONTAINERNAME
+    value: buyingcatalogue-development-helm-documents
+  - group: dev-secrets
+
+steps:
+- task: UseGitVersion@5
+  name: gitversion
+  displayName: Work out version
+  inputs:
+    versionSpec: '5.x'
+
+- task: UseDotNet@2
+  displayName: 'Use DotNet Core $(dotnetVersion)'
+  inputs:
+    version: $(dotnetVersion)
+    includePreviewVersions: false
+
+- bash: |
+    echo "##vso[task.setvariable variable=PBHOSTURL]$(echo "${PBURL#'https://'}")"
```

Comment:
What is setting the PBURL variable in the first place, is this a pipeline variable, or is it in a file somewhere?
Does this mean it needs to know about:
1. PBURL
2. MPSupplierURL
3. MPAuthorityURL

Reply:
It's a pipeline variable, MP Urls should be derived from that URL

Reply:
And is that derivation happening at pipeline variable or code level? I'm not sure I can spot it in this PR

Reply:
pipeline variable level